Created AWS GameLift Scripting doc (#712)
o3de/o3de.org: content/docs/user-guide/gems/reference/aws/aws-gamelift/gem-setup.md

---
linkTitle: Setup
title: AWS GameLift Gem Setup
description: Learn how to set up the AWS GameLift Gem in Open 3D Engine (O3DE).
toc: true
weight: 100
---

> This topic teaches you how to set up the **AWS GameLift** Gem in Open 3D Engine (O3DE) and AWS GameLift in your project. 

Reviewer: ```suggestion
This topic teaches you how to set up the **AWS GameLift** Gem in Open 3D Engine (O3DE) and Amazon GameLift in your project. 
```

---
linkTitle: Setup
title: AWS GameLift Gem Setup
description: Learn how to set up the AWS GameLift Gem in Open 3D Engine (O3DE).
toc: true
weight: 100
---

> This topic teaches you how to set up the **AWS GameLift** Gem in Open 3D Engine (O3DE) and Amazon GameLift in your project. 

Reviewer 2: minor: Not sure we need the part after the and.

## 1. Understand GameLift

> GameLift allows players to connect to your game by creating a game session. For more information, refer to the AWS Amazon GameLift [How Players Connect to Games](https://docs.aws.amazon.com/gamelift/latest/developerguide/game-sessions-intro.html) documentation.

Reviewer: ```suggestion
GameLift allows players to connect to your game by creating a game session. For more information, refer to [How Players Connect to Games](https://docs.aws.amazon.com/gamelift/latest/developerguide/game-sessions-intro.html) in the Amazon GameLift Developer Guide.
```
AWS naming/link text formatting edit, per Pip's review comment.

>    4. Destroy the session when the condition is met. For example, when the last player leaves the session. 

Reviewer: ```
description: "Learn how to set up the AWS GameLift Gem in O3DE"
toc: true
weight: 100
---
This topic teaches you how to set up the AWS GameLift Gem in Open 3D Engine (O3DE) and the AWS GameLift service in your project.
```
I see that we need a couple of changes in the metadata and intro sentence above (not in your changes). Please make these fixes:
* For the description: Drop the quotation marks, use the full name of O3DE, and add a period:
>Learn how to set up the AWS GameLift Gem in Open 3D Engine (O3DE).
* In the intro:
  * Add **bold** formatting to the Gem name first reference.
  * In the intro, change "the AWS GameLift service" to "Amazon GameLift". (Incorrect naming for the AWS service.)

    

## 2. Enable the AWS GameLift Gem

### Dependencies

The AWS GameLift Gem depends on the following Gems:

- [AWS Core Gem](/docs/user-guide/gems/reference/aws/aws-core): Provides the framework to use AWS services in O3DE.
- [Multiplayer Gem](/docs/user-guide/gems/reference/multiplayer/multiplayer/): Provides multiplayer functionality, like connection and hosting, by extending the networking framework.
  

### Enable the AWS GameLift Gem and its dependencies

To enable the AWS GameLift Gem in your project:
1. Open the **Project Manager**. 
2. Open the menu under your project and choose **Edit Project Settings...**.
3. Choose **Configure Gems**.
4. Enable the AWS GameLift Gem and the dependent Gems.

### Include the AWS GameLift Gem static library

You must include the AWS GameLift Gem static library in your project's CMake build target.

1. **(Required)** You must include **Gem::AWSGameLift.Server.Static** as **BUILD_DEPENDENCIES** for your project server target.

> 

Reviewer: In step 2 below this (not in your changes), change "AWS GameLift" ("to make AWS GameLift requests") to "GameLift" to correct the AWS service name.

    ```

> 2. **(Optional)** If you need to make GameLift requests in C++, then you must include **Gem::AWSGameLift.Client.Static** as **BUILD_DEPENDENCIES** for your client target.

Reviewer 2: minor: GameLift service requests in C++

- `JoinSession`
- `LeaveSession`

> The AWS GameLift Gem provides both [C++ APIs](cpp-api/) and [scripting](scripting/) to support developers. You can implement these use cases using either method.  

Reviewer: ```suggestion
The AWS GameLift Gem provides both [C++ APIs](cpp-api/) and [scripting](scripting/). You can implement these use cases using either method.
```
Do we need to say "to support developers" here? Is that redundant, since the reader is already a game dev audience?

## 4. Integrate server

> To establish communication between your server and Amazon GameLift service, you must notify Amazon GameLift that your server is ready and then have your server respond to GameLift notifications.

Reviewer: ```suggestion
To establish communication between your server and GameLift, you must notify GameLift that your server is ready, and then have your server respond to GameLift notifications.
```

> For more details on GameLift server notifications, refer to [C++ API for the AWS GameLift Gem](cpp-api/#server-notifications).

Reviewer: ```suggestion
For more details on GameLift server notifications, refer to the [Server notifications](cpp-api/#server-notifications) section of the AWS GameLift Gem C++ API page.
```

## 5. Set up AWS credentials

> 

Reviewer: In the link text below this (not in your changes), fix the link text ("Configuring AWS Credentials for O3DE") by dropping "for O3DE".

Author: I also dropped the "in the AWS Core section." based on your suggestion from the last review. I forgot to make that change in these updates.

- `JoinSession`
- `LeaveSession`

The AWS GameLift Gem provides both [C++ APIs](cpp-api/) and [scripting](scripting/). You can implement these use cases using either method.

## 4. Integrate server

To establish communication between your server and GameLift, you must notify GameLift that your server is ready, and then have your server respond to GameLift notifications.

For more details on GameLift server notifications, refer to the [Server notifications](cpp-api/#server-notifications) section of the AWS GameLift Gem C++ API page.

## 5. Set up AWS credentials

To work with AWS resources in O3DE you must set up AWS credentials for your users. 

> For more details, read [Configuring AWS Credentials](/docs/user-guide/gems/reference/aws/aws-core/configuring-credentials/).

Reviewer: ```suggestion
For more details, refer to [Configuring AWS Credentials](/docs/user-guide/gems/reference/aws/aws-core/configuring-credentials/).
```

- `JoinSession`
- `LeaveSession`

The AWS GameLift Gem provides both [C++ APIs](cpp-api/) and [scripting](scripting/). You can implement these use cases using either method.

## 4. Integrate server

To establish communication between your server and GameLift, you must notify GameLift that your server is ready, and then have your server respond to GameLift notifications.

For more details on GameLift server notifications, refer to the [Server notifications](cpp-api/#server-notifications) section of the AWS GameLift Gem C++ API page.

## 5. Set up AWS credentials

To work with AWS resources in O3DE you must set up AWS credentials for your users. 

For more details, refer to [Configuring AWS Credentials](/docs/user-guide/gems/reference/aws/aws-core/configuring-credentials/).

{{< note >}} 
This step is only required for developers who perform remote tests and infrastructure builds against the Amazon GameLift service.

> Alternatively, you can test on your local machine using GameLift Local, which does not require AWS credentials. To test against GameLift Local, there is support for the client override. For more information, refer to [AWS GameLift Gem Local Testing](/docs/user-guide/gems/reference/aws/aws-gamelift/local-testing/). 

Reviewer 2: Alternatively, you can test your server's GameLift integration locally using GameLift Local, which does not require AWS credentials or the creation of AWS resources. For more information, refer to [AWS GameLift Gem Local Testing](/docs/user-guide/gems/reference/aws/aws-gamelift/local-testing/).

{{< /note >}} 

## 6. Set up the AWS CLI and AWS CDK

> The AWS GameLift Gem provides a sample AWS Cloud Development Kit (AWS CDK) application that you can use to model and deploy your server to GameLift. To do this, you must have the [AWS Command Line Interface (AWS CLI)](https://aws.amazon.com/cli/) and [AWS CDK](https://aws.amazon.com/cdk/) installed on your local machine.

Reviewer 2: minor: "....that you can use to model Amazon GameLift resources and deploy your server"

> ## 6. Set up AWS CLI and CDK

Reviewer: ```suggestion
## 6. Set up the AWS CLI and AWS CDK
```

> The AWS GameLift provides a sample CDK application that you can use to model and deploy your server to the Amazon GameLift service. To do this, you must have [AWS CLI](https://aws.amazon.com/cli/) and [AWS Cloud Development Kit (CDK)](https://aws.amazon.com/cdk/) installed on your local machine.

Reviewer: ```suggestion
The AWS GameLift Gem provides a sample AWS Cloud Development Kit (AWS CDK) application that you can use to model and deploy your server to GameLift. To do this, you must have the [AWS Command Line Interface (AWS CLI)](https://aws.amazon.com/cli/) and [AWS CDK](https://aws.amazon.com/cdk/) installed on your local machine.
```

---

> Previous Topic: [AWS GameLift Feature Overview](/docs/user-guide/gems/reference/aws/aws-gamelift)  
> Next topic: [C++ API for the AWS GameLift Gem](cpp-api/)

Reviewer: ```suggestion
Previous topic: [AWS GameLift Gem feature overview](/docs/user-guide/gems/reference/aws/aws-gamelift)
Next topic: [AWS GameLift Gem C++ API](cpp-api/)
```
